### PR TITLE
Implement Rate-Limit Handling on Zoom Client Requests

### PR DESCRIPTION
**Pull request by frankmeszaros**
Implement retry logic for zoom requests. Follows a very naive scheme and backs off using the following scheme:

```
wait_for_sec = 2
retry_limit = 5

retry 0: wait 2 sec
retry 1: wait 2 sec 
retry 2: wait 4 sec
retry 3: wait 6 sec
retry 4: wait 8 sec
```

### CODE REVIEW

File: vectrix/clients/zoom.py [hunk 1]
```diff
@@ -37,22 +57,50 @@ def __zoom_headers(self):
...
 
+        except RetryLimitExceeded as e:
+            print(e)
+            raise RetryLimitExceeded()
+
         except Exception as e:
-            logger.exception(e)
-            return None
+            print(e)
+            raise Exception()
```
Comment (samuelbostick) [inline vectrix/clients/zoom.py:103]: Lets continue to `raise e` here instead of a generic `Exception`

**Review by samuelbostick — CHANGES_REQUESTED** [review]
One minor comment.

**Review by samuelbostick — APPROVED** [review]
dope

**Review by samuelbostick — APPROVED** [review]
:fire:

**Review by Lewisecurity — APPROVED** [review]

**Pull request merged**